py/test/selenium/webdriver/common/virtual_authenticator_tests.py

```diff
@@ -144,109 +152,240 @@ def test_add_and_remove_non_resident_credentials(driver, pages):
 @pytest.mark.xfail_safari
 @pytest.mark.xfail_remote
-def test_remove_all_credentials(driver):
-    options = VirtualAuthenticatorOptions()
-    options.has_resident_key = True
+def test_add_resident_credential_not_supported_when_authenticator_uses_u2f_protocol(driver, pages):
+    driver = create_rk_enabled_u2f_authenticator(driver)
+    driver.get(pages.url("virtual-authenticator.html", localhost=True))
 
-    driver.add_virtual_authenticator(options)
-    assert driver.virtual_authenticator_id is not None
+    base64_pk = '''
+    MIGHAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBG0wawIBAQQg8_zMDQDYAxlU-Q
+    hk1Dwkf0v18GZca1DMF3SaJ9HPdmShRANCAASNYX5lyVCOZLzFZzrIKmeZ2jwU
+    RmgsJYxGP__fWN_S-j5sN4tT15XEpN_7QZnt14YvI6uvAgO0uJEboFaZlOEB
+    '''
 
-    credential = Credential.create_non_resident_credential(
+    credential = Credential.create_resident_credential(
         bytearray({1, 2, 3, 4}),
         "localhost",
-        b64decode(BASE64__ENCODED_PK),
+        bytearray({1}),
+        urlsafe_b64decode(base64_pk),
         0,
     )
+    try:
+        driver.add_credential(credential)
+    except InvalidArgumentException:
+        assert True
+    except Exception:
+        assert False
 
-    credential2 = Credential.create_resident_credential(
-        bytearray({1, 2, 3, 4, 5}),
-        "localhost",
-        bytearray({1}),
-        b64decode(BASE64__ENCODED_PK),
-        1,
-    )
+    driver.remove_virtual_authenticator()
 
-    driver.add_credential(credential)
-    assert len(driver.get_credentials()) == 1
 
-    driver.add_credential(credential2)
-    assert len(driver.get_credentials()) == 2
+@pytest.mark.xfail_firefox
+@pytest.mark.xfail_safari
+@pytest.mark.xfail_remote
+def test_get_credentials(driver, pages):
+    driver = create_rk_enabled_ctap2_authenticator(driver)
+    driver.get(pages.url("virtual-authenticator.html", localhost=True))
+
+    # Register a Resident Credential
+    response1 = driver.execute_async_script('''
+    registerCredential({authenticatorSelection: {requireResidentKey: true}})
+    .then(arguments[arguments.length - 1]);
+    ''')
+    assert response1.get('status', '') == 'OK'
+
+    # Register a Non-Resident Credential
+    response2 = driver.execute_async_script(REGISTER_CREDENTIAL)
+    assert response2.get('status', '') == 'OK'
+
+    assert extract_id(response1) != extract_id(response2)
+
+    # Retrieve the two credentials
+    credentials = driver.get_credentials()
+    assert len(credentials) == 2
+
+    credential1, credential2 = None, None
+
+    for credential in credentials:
+        # Using startswith because there can be padding difference '==' or '=' in the end
+        if credential.id.startswith(extract_id(response1)):
+            credential1: Credential = credential
+        elif credential.id.startswith(extract_id(response2)):
+            credential2: Credential = credential
+        else:
+            assert False, "Unknown credential"
+
+    if credential1.is_resident_credential:
```

Comment:
What I meant is that we should have something like

```suggestion
    assert credential1.is_resident_credential == True, "Credential1 should be resident credential"
```

Reply:
I was writing the tests like this. But Flake8 complains about writing tests like this. 
for code 
```python
assert credential1.is_resident_credential == True, "Credential1 should be resident credential"
```
Error
```json
[{
	"resource": "/Users/harmandeepsingh/BrowserStack/selenium/py/test/selenium/webdriver/common/virtual_authenticator_tests.py",
	"owner": "python",
	"code": "E712",
	"severity": 8,
	"message": "comparison to True should be 'if cond is True:' or 'if cond:'",
	"source": "flake8",
	"startLineNumber": 242,
	"startColumn": 47,
	"endLineNumber": 242,
	"endColumn": 47
}]
```

Reply:
Maybe we can skip flake8 in tests to clean the code.

Reply:
@AutomatedTester comparing it with `==` has flake8 problem. But if I compare with `is` then flake8 is happy. So I've changed accordingly.

Reply:
I don't like the idea of skipping flake8 in tests, booleans are singleton values and should be compared with `is` on identity (same with `None`) if not using the built-in implicit truthy checks.  You can just `assert credential1.is_resident_credential` here, checking the value is true is implicit